### PR TITLE
[LS] Improve record field completions

### LINKED ISSUE #40477: [Bug]: Duplicated completion items for fields in mapping  constructor
### Description

Consider the following capture

image

### Steps to Reproduce

```ballerina
service / on new http:Listener {
    resource function get 'order/[string orderId]() returns http:Ok|http:NotFound {
        // Write your code here
        if !orders.hasKey(orderId) {
            return <http:NotFound>{};
        }
        return <http:Ok>{};
    }
}

```

### Affected Version(s)

2201.5.0

### OS, DB, other environment details and versions

_No response_

### Related area

-> Editor

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by malinthar**
## Purpose
$subject

Fixes #40477

## Approach
When there are multiple fields with the same name and same type, show a single completion item for the particular field having a detail of `Type1.field1 | Type2.field1`.

## Samples
image

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/RecordUtil.java [hunk 1]
```diff
@@ -50,45 +51,59 @@ public class RecordUtil {
...
+
+        //Assuming the record fields list is not empty at this level
+        RecordField recordField = fields.get(0);
+        String name = recordFieldIdentifier.getName();
+        SnippetContext snippetContext = new SnippetContext();
+        String insertText = getRecordFieldCompletionInsertText(recordField.getFieldSymbol(), snippetContext);
+
+        String detail = fields.stream().map(field -> {
+            RawTypeSymbolWrapper<RecordTypeSymbol> wrapper = field.getTypeSymbolWrapper();
+            if (recordField.getTypeSymbolWrapper().getRawType().getName().isPresent()) {
```
Comment (KavinduZoysa) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/RecordUtil.java:81]: Shall we add a test case for this?

**Comment by KavinduZoysa** [comment]
```ballerina
type INT int;

type Foo record {|
    int i;
|};

type Bar record {|
    INT i;
|};

public function main() {
    Foo|Bar fb = {}
}
```
For above case we are getting two suggestions, is that expected?

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/RecordUtil.java [hunk 2]
```diff
@@ -50,45 +51,59 @@ public class RecordUtil {
...
+    }
 
-            String detail;
-            if (wrapper.getRawType().getName().isPresent()) {
-                detail = NameUtil.getModifiedTypeName(context, wrapper.getRawType()) + "." + name;
+    private static LSCompletionItem getRecordFieldCompletionItem(BallerinaCompletionContext context,
+                                                 RecordField.RecordFieldIdentifier recordFieldIdentifier,
+                                                 List<RecordField> fields) {
+
+        //Assuming the record fields list is not empty at this level
```
Comment (mindula) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/RecordUtil.java:73]: Let's check if the fields list is empty

File: language-server/modules/langserver-core/src/test/resources/completion/expression_context/config/mapping_constructor_expr_ctx_config7.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+  "position": {
+    "line": 19,
+    "character": 35
+  },
+  "source": "expression_context/source/mapping_constructor_expr_ctx_source7.bal",
+  "description": "",
```
Comment (mindula) [inline language-server/modules/langserver-core/src/test/resources/completion/expression_context/config/mapping_constructor_expr_ctx_config7.json:7]: Add a description

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/RecordUtil.java (hunk 1 above)
Comment (malinthar) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/RecordUtil.java:81]: We can remove this if condition

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/RecordUtil.java (hunk 2 above)
Comment (KavinduZoysa) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/RecordUtil.java:73]: I think this can't be empty.

Comment (malinthar) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/RecordUtil.java:73]: Yes at this point the completion list shouldn't be empty

**Comment by malinthar** [comment]
> ```ballerina
> type INT int;
> 
> type Foo record {|
>     int i;
> |};
> 
> type Bar record {|
>     INT i;
> |};
> 
> public function main() {
>     Foo|Bar fb = {}
> }
> ```
> 
> For above case we are getting two suggestions, is that expected?

Yes that's expected! We are showing a single completion item for multiple fields only when the type of the fields is exactly the same.

**Review by KavinduZoysa — APPROVED** [review]

**Review by mindula — APPROVED** [review]

**Pull request merged**